app/controllers/admin/products_v3_controller.rb

```diff
@@ -68,6 +68,29 @@ def destroy_variant
       end
     end
 
+    def clone
+      @product = Spree::Product.find(params[:id])
+      status = :ok
+
+      begin
+        @cloned_product = @product.duplicate
+        flash.now[:success] = t('.success')
+
+        @product_index = "-#{@cloned_product.id}"
+        @producer_options = producers
+        @category_options = categories
+        @tax_category_options = tax_category_options
+      rescue StandardError => _e
```

Comment:
I somehow missed this in my last review. We need to avoid rescuing `StandardError` because it could be hiding other important errors. Instead we can rescue a specific error.

And we should have a test to cover the error case too. Would you be able to add one?

If the `.duplicate` method returns a false on error, perhaps we should branch off that (that's what we normally do when creating/updating in controllers). Alternatively, you could change it to rescue the specific error.

Reply:
@dacook - The duplicate method is saving the cloned product with `.save!`. I used the `StandardError`, to rescue for that the failure of `.save!` 😅. After saving, it returns the product:

https://github.com/chahmedejaz/openfoodnetwork/blob/3afb3f5ba05b7565dd32c68b638d152ee2690ec4/lib/spree/core/product_duplicator.rb#L12-L16

I think branching off would be better to handle this case, but that would cause changes in multiple places. Is that okay?

> And we should have a test to cover the error case too. Would you be able to add one?

Sure, I'll add that no worries.

Reply:
Hi @chahmedejaz , thanks for the quick turnaround. I discussed with Maikel and we agreed it was heading in a good direction, but decided to first update specs to understand the error case better.
After doing that, it was clearer and I found that an exception could still occur due to invalid data on the variants. 

It looked difficult to handle that, so I decided it's simpler to keep the exception handling as it is, but with more specific exceptions. Does that look ok to you?

Reply:
Looks good @dacook - Thank you so much for the help here ⭐🙂